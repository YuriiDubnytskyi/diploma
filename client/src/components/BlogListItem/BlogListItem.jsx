import React from "react";
import "./BlogListIte.scss";
import { useHistory } from "react-router-dom";
import { Link } from "react-router-dom";

const BlogListItem = ({ title, description, id, time, imgSrc }) => {
    const history = useHistory();
    return (
        <div className="blog__item blog-item">
            <figure class="snip1527">
                <div class="image">
                    <img src={imgSrc} alt={title} />
                </div>
                <figcaption>
                    <div class="date">
                        <span class="day">{new Date(time).getDate()}</span>
                        <span class="month">{new Date(time).getMonth() + 1}</span>
                    </div>
                    <h3>{title}</h3>
                    <p>{description}</p>
                </figcaption>
                <Link to={`/newsID/${id}`}></Link>
            </figure>
            {/* <div className="blog-item-image"></div>
            <div className="blog-item-info item-info" onClick={() => history.push("/newsID/" + id)}>
                <h3 className="item-info-title">{title}</h3>
                <p className="item-info-description">
                    {description}
                    Description Description Description Description Description Description Description Description
                    Description Description Description Description Description Description Description Description
                    Description Description Description Description
                </p>
            </div> */}
        </div>
    );
};

BlogListItem.whyDidYouRender = true;
export default React.memo(BlogListItem);
