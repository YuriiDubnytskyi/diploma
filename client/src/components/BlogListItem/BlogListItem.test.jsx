import React from "react";
import BlogListItem from "./BlogListItem";

const setUp = (props) => shallow(<BlogListItem {...props} />);

describe("Test HTML in BlogListItem", () => {
    it("Should contain title, description", () => {
        const component = setUp({ title: "News", description: "Desc" });
        expect(component).toMatchSnapshot();
    });
});
