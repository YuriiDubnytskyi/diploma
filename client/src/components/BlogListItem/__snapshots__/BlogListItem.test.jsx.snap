// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HTML in BlogListItem Should contain title, description 1`] = `
<div
  className="blog__item blog-item"
>
  <figure
    class="snip1527"
  >
    <div
      class="image"
    >
      <img
        alt="News"
      />
    </div>
    <figcaption>
      <div
        class="date"
      >
        <span
          class="day"
        >
          NaN
        </span>
        <span
          class="month"
        >
          NaN
        </span>
      </div>
      <h3>
        News
      </h3>
      <p>
        Desc
      </p>
    </figcaption>
    <Link
      to="/newsID/undefined"
    />
  </figure>
</div>
`;
